### DISCUSSION TITLE
How do hooks work?

### DISCUSSION THREAD

**Discussion by ervinpopescu** [qtile/qtile#4270] [Q&A] [answered]
I'm looking through `libqtile.hook` and all I can see is stars xD. Any help? 🆘

**Comment by elParaguayo** [comment, marked as answer]
Basically....
- When you subscribe for an event the hooked function is stored in a dictionary where the event name is the key and the list of callbacks is the value
- When an event is triggered, qtile looks for all functions subscribed to that event and fires them one by one.

**Reply by ervinpopescu** [reply]
Thanks 😄. It was actually easier than I thought. My initial thought was that there was something in the manager that controlled all of the hooks somehow, but now I get that there is no need for that.
